From a7df875788d7303639123771af1f4238e207f442 Mon Sep 17 00:00:00 2001
Message-ID: <a7df875788d7303639123771af1f4238e207f442.1730304104.git.sam@gentoo.org>
From: Sam James <sam@gentoo.org>
Date: Wed, 30 Oct 2024 15:59:01 +0000
Subject: [PATCH] Revert "Match: Simplify (x != 0 ? x + ~0 : 0) to (x - x !=
 0)."

This reverts commit 4af8db3eca12b2db3753ce4b098cbd0ae32b4796.

Bug: https://gcc.gnu.org/PR117363
---
 gcc/match.pd                               | 10 ---------
 gcc/testsuite/gcc.dg/tree-ssa/phi-opt-44.c | 26 ----------------------
 gcc/testsuite/gcc.dg/tree-ssa/phi-opt-45.c | 26 ----------------------
 3 files changed, 62 deletions(-)
 delete mode 100644 gcc/testsuite/gcc.dg/tree-ssa/phi-opt-44.c
 delete mode 100644 gcc/testsuite/gcc.dg/tree-ssa/phi-opt-45.c

diff --git a/gcc/match.pd b/gcc/match.pd
index c851ac56e37c..809c717bc862 100644
--- a/gcc/match.pd
+++ b/gcc/match.pd
@@ -3391,16 +3391,6 @@ DEFINE_INT_AND_FLOAT_ROUND_FN (RINT)
   }
   (if (wi::eq_p (sum, wi::uhwi (0, precision)))))))
 
-/* The boundary condition for case 10: IMM = 1:
-   SAT_U_SUB = X >= IMM ? (X - IMM) : 0.
-   simplify (X != 0 ? X + ~0 : 0) to (X - X != 0).  */
-(simplify
- (cond (ne@1 @0 integer_zerop)
-       (nop_convert? (plus (nop_convert? @0) integer_all_onesp))
-       integer_zerop)
- (if (INTEGRAL_TYPE_P (type))
-   (minus @0 (convert @1))))
-
 /* Signed saturation sub, case 1:
    T minus = (T)((UT)X - (UT)Y);
    SAT_S_SUB = (X ^ Y) & (X ^ minus) < 0 ? (-(T)(X < 0) ^ MAX) : minus;
diff --git a/gcc/testsuite/gcc.dg/tree-ssa/phi-opt-44.c b/gcc/testsuite/gcc.dg/tree-ssa/phi-opt-44.c
deleted file mode 100644
index 962bf0954f62..000000000000
--- a/gcc/testsuite/gcc.dg/tree-ssa/phi-opt-44.c
+++ /dev/null
@@ -1,26 +0,0 @@
-/* { dg-do compile } */
-/* { dg-options "-O2 -fdump-tree-phiopt1" } */
-
-#include <stdint.h>
-
-uint8_t f1 (uint8_t x)
-{
-  return x >= (uint8_t)1 ? x - (uint8_t)1 : 0;
-}
-
-uint16_t f2 (uint16_t x)
-{
-  return x >= (uint16_t)1 ? x - (uint16_t)1 : 0;
-}
-
-uint32_t f3 (uint32_t x)
-{
-  return x >= (uint32_t)1 ? x - (uint32_t)1 : 0;
-}
-
-uint64_t f4 (uint64_t x)
-{
-  return x >= (uint64_t)1 ? x - (uint64_t)1 : 0;
-}
-
-/* { dg-final { scan-tree-dump-not "goto" "phiopt1" } } */
diff --git a/gcc/testsuite/gcc.dg/tree-ssa/phi-opt-45.c b/gcc/testsuite/gcc.dg/tree-ssa/phi-opt-45.c
deleted file mode 100644
index 62a2ab631846..000000000000
--- a/gcc/testsuite/gcc.dg/tree-ssa/phi-opt-45.c
+++ /dev/null
@@ -1,26 +0,0 @@
-/* { dg-do compile } */
-/* { dg-options "-O2 -fdump-tree-phiopt1" } */
-
-#include <stdint.h>
-
-int8_t f1 (int8_t x)
-{
-  return x != 0 ? x - (int8_t)1 : 0;
-}
-
-int16_t f2 (int16_t x)
-{
-  return x != 0 ? x - (int16_t)1 : 0;
-}
-
-int32_t f3 (int32_t x)
-{
-  return x != 0 ? x - (int32_t)1 : 0;
-}
-
-int64_t f4 (int64_t x)
-{
-  return x != 0 ? x - (int64_t)1 : 0;
-}
-
-/* { dg-final { scan-tree-dump-not "goto" "phiopt1" } } */
-- 
2.47.0

